user-lock-manager-w-keypad-support.smartapp.groovy

```diff
@@ -1293,3 +1340,154 @@ def populateDiscovery(codeData, lock) {
   }
   state."lock${lock.id}".codes = codes
 }
+
+private String getPIN() {
+	return settings.pin.value.toString().padLeft(4,'0')
+}
+
+def alarmStatusHandler(event) {
+	log.debug "Keypad manager caught alarm status change: "+event.value
+    if (event.value == "off") keypad?.setDisarmed()
+    else if (event.value == "away") keypad?.setArmedAway()
+    else if (event.value == "stay") keypad?.setArmedStay()
+}
+
+private sendSHMEvent(String shmState) {
+	def event = [name:"alarmSystemStatus", value: shmState, 
+    			displayed: true, description: "System Status is ${shmState}"]
+                log.debug "test ${event}"
+    sendLocationEvent(event)
+}
+
+private execRoutine(armMode) {
+	if (armMode == 'away') location.helloHome?.execute(settings.armRoutine)
+    else if (armMode == 'stay') location.helloHome?.execute(settings.stayRoutine)
+    else if (armMode == 'off') location.helloHome?.execute(settings.disarmRoutine)    
+}
+
+def codeEntryHandler(evt) {
+	//do stuff
+    log.debug "Caught code entry event! ${evt.value.value}"
+    
+    def codeEntered = evt.value as String
+    
+    def data = evt.data as String
+    def armMode = ''
+    def currentarmMode = keypad.currentValue("armMode")
+    def changedMode = 0
+    
+    if (data == '0') {
+    	armMode = 'off'
+    }
+    else if (data == '3') {
+    	armMode = 'away'
+    }
+    else if (data == '1') {
+    	armMode = 'stay'
+    }
+    else if (data == '2') {
+    	armMode = 'stay' //Currently no separate night mode for SHM, set to 'stay'
+    }
+    else {
+    	log.error "${app.label}: Unexpected arm mode sent by keypad!: "+data
+        return []
+        }
+        
+        def i = 0
+        def message = " "
+        
+        i = settings.maxUsers
+    while (i > 0) {
```

Comment:
this section is just looping through once for each user in the app. It starts at the max number and counts down until it hits 0. When it matches a code with the code entered it sets the variable to 0. I'm not familiar with the 'for' loop which is why I just used the 'while'. It works properly as it is right now.